heudiconv/workflows/run.py

```diff
@@ -0,0 +1,343 @@
+        establishing a hierarchy. Setting to "unknown" will skip that
+        dataset. Default is None.
+    conv_outdir : str or None, optional
+        Output directory for converted files. By default this is
+        identical to --outdir. This option is most useful in
+        combination with 'anon_cmd'. Default is None.
+    anon_cmd : str or None, optional
+        Command to run to convert subject IDs used for DICOMs to
+        anonymized IDs. Such command must take a single argument and
+        return a single anonymized ID. Also see 'conv_outdir'. Default is None.
+    heuristic : str or None, optional
+        Name of a known heuristic or path to the Python script containing
+        heuristic. Default is None.
+    with_prov : bool, optional
+        Store additional provenance information. Requires python-rdflib.
+        Default is False.
+    session : str or None, optional
+        Session for longitudinal study_sessions. Default is None.
+    bids_options : str or None, optional
+        Flag for output into BIDS structure. Can also take BIDS-
+        specific options, e.g., --bids notop. The only currently
+        supported options is "notop", which skips creation of
+        top-level BIDS files. This is useful when running in batch
+        mode to prevent possible race conditions. Default is None.
+    overwrite : bool, optional
+        Overwrite existing converted files. Default is False.
+    datalad : bool, optional
+        Store the entire collection as DataLad dataset(s). Small files
+        will be committed directly to git, while large to annex. New
+        version (6) of annex repositories will be used in a "thin"
+        mode so it would look to mortals as just any other regular
+        directory (i.e. no symlinks to under .git/annex). For now just
+        for BIDS mode. Default is False.
+    debug : bool, optional
+        Do not catch exceptions and show exception traceback. Default is False.
+    command : {'heuristics', 'heuristic-info', 'ls', 'populate-templates',
+               'sanitize-jsons', 'treat-jsons', None}, optional
+        Custom action to be performed on provided files instead of regular
+        operation. Default is None.
+    grouping : {'studyUID', 'accession_number', 'all', 'custom'}, optional
+        How to group dicoms. Default is 'studyUID'.
+    minmeta : bool, optional
+        Exclude dcmstack meta information in sidecar jsons. Default is False.
+    random_seed : int or None, optional
+        Random seed to initialize RNG. Default is None.
+    dcmconfig : str or None, optional
+        JSON file for additional dcm2niix configuration. Default is None.
+    queue : {'SLURM', None}, optional
+        Batch system to submit jobs in parallel. Default is None.
+    queue_args : str or None, optional
+        Additional queue arguments passed as single string of space-separated
+        Argument=Value pairs. Default is None.
+    """
```

Comment:
I don't know an "easy" way around it ATM, may be you would find some, but such duplication of docstrings in here and in cmdline argparse setup eventually would diverge.  I don't know an "easy" way since most of the time then people indeed either duplicate like you have done, or come up with some evolved way to generate both cmdline and python interfaces from the same specification (like we did in datalad, and I saw similar ones elsewhere).

In principle it should be possible for many (if not all -- didn't do matching which one is pertinent only here and not cmdline) to take argparse specification and use it to populate a docstring here, and while at it may be even verify consistency of the function.  It feels like a common problem, so may be someone already came up with a similar solution?  I was looking for the one for `click` based interfaces but found none... could you check @tsalo around to see if smth like that was already done, or may be just identify a showstopper which would forbid it?

Reply:
I tried looking around for tools to synchronize argument parsers with associated functions' docstrings, but couldn't find anything. If it's any consolation, we have the same separation in `tedana` and, while there's no automated control, keeping the function docstring up-to-date with the CLI is something we've been able to do. I could also add a line in the function description telling users to refer to the command-line help for the *definitive* argument descriptions.

Reply:
wouldn't the default click behavior work. see the top example here: https://click.palletsprojects.com/en/7.x/

Reply:
for any developer it should be clear from looking at the decorators.

Reply:
I think that that would take whatever's in the function docstring and make it available in the command line help, but when you try `help(function)` in Python, you get a very long and unrelated string for `click.Command` instead of the function. I don't think click would help with within-Python documentation.